### PR TITLE
[RALP-4730] Update visual appearance of badges

### PR DESCRIPTION
**Pull request by gergana-st-ivanova**
Update visual appearance of Badges
- SwiftUI part
- UIKit part

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `README.md`
+ [ ] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack-SwiftUI/Badge/Classes/BPKBadge.swift [hunk 1]
```diff
@@ -60,14 +60,29 @@ public struct BPKBadge: View {
         return result
     }
     
+    public func createBadgeIconView(icon: BPKIcon?) -> BPKIconView? {
+        guard let badgeIcon = icon else {
+            switch style {
+            case .normal, .strong, .inverse, .outline, .brand:
+                return nil
```
Comment (gert-janvercauteren) [inline Backpack-SwiftUI/Badge/Classes/BPKBadge.swift:67]: In this case I would use 'default' to return nil. As only the 3 'status' badges receive icons

File: screenshots/iPhone SE (3rd generation)-swiftui_badge___default_lm.png
Comment (gert-janvercauteren) [inline screenshots/iPhone SE (3rd generation)-swiftui_badge___default_lm.png:1]: We probably need to rename these so they match the original. Not quite sure what made them differently named :(

**Review by gert-janvercauteren — COMMENTED** [review]
Great work! Just a few nitpicks :D

**Comment by gergana-st-ivanova** [comment]
> Great work! Just a few nitpicks :D
> Can you make sure the snapshot tests that weren't changed are not changed in this PR as well? (Calendar etc)
Thank you, @gert-janvercauteren! These (Calendar and Flow View) are using the badges so they also had to be recreated for the tests to pass I believe.

File: screenshots/iPhone SE (3rd generation)-swiftui_badge___default_lm.png
Comment (gergana-st-ivanova) [reply screenshots/iPhone SE (3rd generation)-swiftui_badge___default_lm.png:1]: Which one is the original? [screenshots/iPhone-badge___all_dm.png](https://github.com/Skyscanner/backpack-ios/pull/1873/files/36a1f3ec84c25d605662e393e46aed05968ec52c#diff-f7372b776198bf12fd7999d56612ba9149f0e00da955360c7c4188ce77f39e61) this was the original I think but there is a file with that name created. Could those be newly generated ones?

**Review by gert-janvercauteren — COMMENTED** [review]
Reviewed one more time, let me know if anything is unclear

**Review by brunomdac — COMMENTED** [review]
Looks great to me 👏  However before merging, as Gert pointed out, it'd be good to remove the snapshots that should not have changed as part of these changes (calendar and stackView snapshots at least)

File: Backpack-SwiftUI/Tests/FlowStackView/__Snapshots__/BPKFlowStackViewTests/test_flowStackView_center.dark-mode.png
Comment (brunomdac) [inline Backpack-SwiftUI/Tests/FlowStackView/__Snapshots__/BPKFlowStackViewTests/test_flowStackView_center.dark-mode.png:1]: For example, the resolution change in this snapshot, where I believe our changes should have no impact.

Comment (marianeum) [reply Backpack-SwiftUI/Tests/FlowStackView/__Snapshots__/BPKFlowStackViewTests/test_flowStackView_center.dark-mode.png:1]: this one makes sense as the badge is being used in the example

**Review by marianeum — APPROVED** [review]
snapshot changes look good!

**Review by brunomdac — APPROVED** [review]
Looks great, great job! 👏

**Pull request merged**